src/components/data-loader.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { v1 } from 'uuid';
+import '../styles/components/data-loader.scss';
+
+const scrollOffsetPercentage = 10;
+const scrollOffsetFactor = 1 + scrollOffsetPercentage / 100;
+
+const isNotScrollable = (node) => {
+  const { scrollHeight, offsetHeight } = node.current;
+  return scrollHeight <= offsetHeight * scrollOffsetFactor;
+};
+
+const withDataLoader = BaseComponent => (props) => {
+  const {
+    onLoadMoreItems, hasMoreData, data, loaderComponent = 'Loading...',
+  } = props;
+
+  const [loading, setLoading] = useState(false);
+  const [loadMoreItems, setLoadMoreItems] = useState(false);
+  const ref = useRef();
+  const loadingMessageId = v1();
+
+  useEffect(() => {
+    if (hasMoreData && isNotScrollable(ref)) {
+      // Note: @dan, commenting the following line
```

Comment:
There shouldn't be personal messages/comments in the code for other team members. If the functionality can be tested and is still correct after `setLoading(true)` is removed, then it should be removed. If it can't be indicated if this is a dead code or not, the comment can be written as: `// Is this function call still necessary?`.